application/src/main/java/org/thingsboard/server/service/queue/processing/TbRuleEngineProcessingStrategyFactory.java

```diff
@@ -101,6 +104,15 @@ public TbRuleEngineProcessingDecision analyze(TbRuleEngineProcessingResult resul
                     if (retrySuccessful) {
                         result.getSuccessMap().forEach(toReprocess::put);
                     }
+                    if (CollectionUtils.isEmpty(toReprocess)) {
+                        log.debug("[{}] Stopping the reprocessing logic due to reprocessing map is empty", queueName);
+                        if (retryFailed) {
```

Comment:
There are very few use cases when software engineer should use 'probably' in their code or logs :-)

Please replace with something like this:
```
 if (retryTimeout) {
    result.getPendingMap().forEach(toReprocess::put);
} else if (log.isDebugEnabled() && !result.getFailedMap().isEmpty()) {
  log.debug("[{}] Skipped {} failed messages due to the processing strategy configuration", queueName, result.getFailedMap().size());
}

```

Reply:
Agree! Updated.